Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -35,11 +36,95 @@ internal static void EnsureMSBuildForUnity()
             if (!File.Exists(manifestPath))
             {
                 Debug.LogError($"Package manifest file ({manifestPath}) could not be found.");
-                return;
+                return null;
+            }
+
+            return manifestPath;
+        }
+
+        /// <summary>
+        /// Reports whether or not the appropriate version of MSBuild for Unity is specified
+        /// in the Unity Package Manager manifest.
+        /// </summary>
+        /// <returns>
+        /// True if an appropriate verson of MS Build for Unity is configured in the manifest, otherwise false.
+        /// </returns>
+        internal static bool IsMSBuildForUnityEnabled()
+        {
+            string manifestPath = GetPackageManifestFilePath();
+            if (string.IsNullOrWhiteSpace(manifestPath))
+            {
+                return false;
+            }
+
+            // Load the manifest file.
+            string manifestFileContents = File.ReadAllText(manifestPath);
+            if (string.IsNullOrWhiteSpace(manifestFileContents))
+            {
+                return false;
+            }
+
+            // Read the package manifest a line at a time.
+            bool msBuildFound = false;
+            bool isAppropriateVersion = false;
+            Version minVersion = Version.Parse(MSBuildPackageVersion);
+            using (FileStream manifestStream = new FileStream(manifestPath, FileMode.Open, FileAccess.Read))
```

Comment:
I think that we've already opened the file and read all of its contents at this point right? (i.e. manifestFileContents?)

If we're trying to be optimizing for stuff it might just be simple enough to read the stat of the file (i.e. size) and not do anything if it's empty. Otherwise there's no reason to read the same file multiple times. (i.e. in trying to be more efficient/smarter here, we're actually less efficient because we're reading the same file twice)

Reply:
we actually have always read the file more than once...